pinc/Round.inc

```diff
@@ -563,3 +563,61 @@ function is_formatting_round($round)
 {
     return startswith($round->id, 'F');
 }
+
+/**
+ * From an array & key, get a Round object matching the ID
+ *
+ * @param array $arr
+ * @param string $key
+ * @param mixed $default
+ * @param bool $allownull
+ *
+ * @throws InvalidArgumentException
+ */
+function get_round_param($arr, $key, $default = null, $allownull = false)
+{
+    global $Round_for_round_id_;
+    // sanity checks on the args
+    assert(is_array($arr));
+    assert(!is_array($key));
+    assert(is_null($default) || is_string($default));
+    if (!is_null($default)) {
+        assert(!$allownull);
+    }
+    if (isset($arr[$key])) {
+        $s = $arr[$key];
+        if (!is_string($s)) {
+            throw new InvalidArgumentException(sprintf(
+                _("Parameter '%1\$s' is not a valid type"),
+                $key
+            ));
+        }
+        // Trim whitespace from both ends of the string.
+        $s = trim($s);
+        if ($s == '' && $allownull) {
+            return null;
+        }
+
+        $round = $Round_for_round_id_[$s] ?? null;
+        if ($round) {
+            return $round;
+        } else {
```

Comment:
minor, but no else is needed since you returned in the associated if statement. Depending on your preference, removing the code may make it easier to read. Same below if you decide to make any change.

Reply:
I really need to read up some on different thoughts on this, because I'm always torn. Is the "right" thing to return early (like this) and keep the else for clarity? Should we return early and not keep the else? Instead of returning should we do a variable assignment and only return once at the end of the function? I've seen (and done) all of that, admittedly inconsistently. My gut says that this is probably a bikeshed problem but with some pros/cons to each approach that are worth considering.

In this case I want to mirror the other validation functions in `misc.inc` which keeps the else after the return.

Reply:
oh completely a bikeshedding event. I think what we have is completely acceptable and maybe preferred depending on your thoughts.